kotest-extensions/kotest-extensions-testcontainers/src/jvmTest/kotlin/io/kotest/extensions/testcontainers/TestcontainerKotestIntegrationPerSpec.kt

```diff
@@ -0,0 +1,17 @@
+package io.kotest.extensions.testcontainers
+
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.shouldBe
+
+class TestcontainerKotestIntegrationPerSpec : StringSpec({
+   val testStartable = TestStartable()
+   listeners(testStartable.perSpec())
```

Comment:
In master I have changed listener to return the type passed in.
So we could do

```
val startable = listener(TestStartable().perSpec())
```

Another option is 
```
val startable = TestStartable().apply { listener(this) }
```

And a final option would be to change the extension method to do the registration itself, then you could do

```
val startable = TestStartable().perSpec()
```

Reply:
We could add a function to `TestConfiguration` like this:
```
   fun <T : TestListener> T.register(): T {
      listener(this)
      return this
   }
```

Then you could just do 

```
val startable = TestStartable().register()
```

But it does mean that you would have to have some way of deciding if the test startable is a per spec or per test. Perhaps

```
val startable = TestStartable().perSpec.register()
```

Reply:
```val startable = TestStartable().perSpec.register()```
Looks good to me.